OpenRA.Mods.Common/Traits/LimitedAmmo.cs

```diff
@@ -17,58 +17,121 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Actor has a limited amount of ammo, after using it all the actor must reload in some way.")]
 	public class LimitedAmmoInfo : ITraitInfo
 	{
+		[Desc("Used to link Armaments to this AmmoPool.")]
+		public readonly string AmmoPoolName = "primary";
+
+		[Desc("How much limited ammo does this pool contain when fully loaded.")]
 		public readonly int Ammo = 0;
 
-		[Desc("Defaults to value in Ammo.")]
-		public readonly int PipCount = 0;
+		[Desc("Initial ammo the actor is created with. Defaults to Ammo.")]
+		public readonly int InitialAmmo = -1;
+
+		[Desc("Defaults to value in Ammo. 0 means no visible pips.")]
+		public readonly int PipCount = -1;
 
+		[Desc("PipType to use for loaded ammo.")]
 		public readonly PipType PipType = PipType.Green;
+
+		[Desc("PipType to use for empty ammo.")]
 		public readonly PipType PipTypeEmpty = PipType.Transparent;
 
-		[Desc("Time to reload measured in ticks.")]
+		[Desc("How much ammo is reloaded after a certain period.")]
+		public readonly int ReloadCount = 0;
+
+		[Desc("How much ammo is reloaded after a certain period.")]
+		public readonly string RearmSound = null;
+
+		[Desc("Time to reload on airfield etc. measured in ticks.")]
 		public readonly int ReloadTicks = 25 * 2;
 
-		public object Create(ActorInitializer init) { return new LimitedAmmo(this); }
+		[Desc("Whether or not ammo is replenished on its own.")]
+		public readonly bool SelfReloads = false;
+
+		[Desc("Time to reload automatically measured in ticks.")]
+		public readonly int SelfReloadTicks = 25 * 2;
+
+		[Desc("Whether or not reload counter should be reset when ammo has been fired.")]
+		public readonly bool ResetOnFire = false;
+
+		public object Create(ActorInitializer init) { return new LimitedAmmo(init.Self, this); }
 	}
 
-	public class LimitedAmmo : INotifyAttack, IPips, ISync
+	public class LimitedAmmo : INotifyAttack, IPips, ITick, ISync
 	{
-		[Sync] int ammo;
-		LimitedAmmoInfo info;
+		[Sync] public int CurrentAmmo;
+		[Sync] public int RemainingTicks;
+		public int PreviousAmmo;
+		readonly LimitedAmmoInfo info;
 
-		public LimitedAmmo(LimitedAmmoInfo info)
+		public LimitedAmmo(Actor self, LimitedAmmoInfo info)
 		{
-			ammo = info.Ammo;
+			if (info.InitialAmmo < info.Ammo && info.InitialAmmo >= 0)
+				CurrentAmmo = info.InitialAmmo;
+			else CurrentAmmo = info.Ammo;
+
+			RemainingTicks = info.ReloadTicks;
+			PreviousAmmo = GetAmmoCount();
 			this.info = info;
 		}
 
-		public bool FullAmmo() { return ammo == info.Ammo; }
-		public bool HasAmmo() { return ammo > 0; }
+		public string AmmoPoolName { get { return info.AmmoPoolName; } }
+		public string RearmSound { get { return info.RearmSound; } }
+		public bool ReloadsAutomatically() { return info.SelfReloads == true; }
```

Comment:
`== true` is unnecessary.

All of these "getters" for the *Info properties are unnecessary bloat, just make the Info public.